### PR TITLE
chore: move balance type and anchors mod to chain_data

### LINKED ISSUE #128: Merge `tx_data_traits.rs` into `chain_data.rs`.
Remove the `tx_data_traits.rs` file and move all of it's contents into `chain_data.rs`. Also move the `Balance` struct into `chain_data.rs`.

See https://discord.com/channels/753336465005608961/753367451319926827/1161983570538147932

### PR DESCRIPTION
**Pull request by startup-dreamer**
### Description

Tries to Fix: bitcoindevkit/bdk_wallet#128 

Moved anchore mod (dropped `tx_data_traits.rs`) and `Balance` type to `chain_data.rs` as suggested in [This comment](https://github.com/bitcoindevkit/bdk_wallet/issues/128)

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

### CODE REVIEW

File: crates/chain/src/chain_data.rs [hunk 1]
```diff
@@ -40,6 +41,62 @@ impl<A: Anchor> ChainPosition<A> {
     }
 }
 
+/// Balance, differentiated into various categories.
```
Comment (LLFourn) [inline crates/chain/src/chain_data.rs]: `Balance` doesn't really make sense here can we just put it in `balance.rs` and `pub use balance::*` from `lib.rs`.

**Review by LLFourn — CHANGES_REQUESTED** [review]

**Comment by startup-dreamer** [comment]
Note (unrelated to PR): GitHub was acting weird for me, that's why this took so long, and I took this issue. I wanted to complete this. Apologies for the delay, @ValuedMammal. Let me know if there are any changes to make. Everything is working fine now.

**Comment by ValuedMammal** [comment]
Approach NACK

**Comment by startup-dreamer** [comment]
> Approach NACK

Got it also it's been stale for soo long closing this PR.

**Pull request closed without merging**